### PR TITLE
Rchain 3430 remove error handler and effect

### PR DESCRIPTION
**Pull request by rabbitonweb**
## Overview
ErrorHandler is mostly not used in comm (or used poorly), but that constraint pushes a lot complexity to other layers that use comm. Once ErrorHandler is removed, a lot of that complexity will be removed

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3430

### CODE REVIEW

**Comment by lukasz-golebiewski** [comment]
Not sure if the last commit is necessary. Keeping the effect aliased would mean that less changes are needed when the time comes to move away from monix Task

File: comm/src/main/scala/coop/rchain/comm/rp/HandleMessages.scala [hunk 1]
```diff
@@ -75,16 +75,21 @@ object HandleMessages {
...
-      _        <- TransportLayer[F].send(peer, response) >>= ErrorHandler[F].fromEither
-      _        <- Log[F].info(s"Responded to protocol handshake request from $peer")
-      _        <- ConnectionsCell[F].flatModify(_.addConnAndReport[F](peer))
+      resErr   <- TransportLayer[F].send(peer, response)
+      _ <- resErr.fold(
+            kp(().pure[F]),
+            kp(
+              Log[F].info(s"Responded to protocol handshake request from $peer") >>
+                ConnectionsCell[F].flatModify(_.addConnAndReport[F](peer))
+            )
```
Comment (ArturGajowy) [inline comm/src/main/scala/coop/rchain/comm/rp/HandleMessages.scala]: I'm sure we can have it simpler than `.fold(kp(().pure[F]), X)`. I think `.map(X)`?

File: node/src/main/scala/coop/rchain/node/diagnostics/client/Runtime.scala [hunk 2]
```diff
@@ -35,18 +35,6 @@ object Runtime {
...
     } yield ()
 
-    for {
-      result <- program.attempt
-      _ <- result match {
-            case Left(ex) => ConsoleIO[F].println(s"Error: ${processError(ex).getMessage}")
-            case _        => ().pure[F]
-          }
-    } yield ()
-  }
```
Comment (ArturGajowy) [inline node/src/main/scala/coop/rchain/node/diagnostics/client/Runtime.scala]: This seems like an unrelated change for `Remove everywhere ErrorHandler as typeclass constraint`. Is the delete intentional?

File: node/src/main/scala/coop/rchain/node/NodeEnvironment.scala [hunk 3]
```diff
@@ -20,24 +20,24 @@ object NodeEnvironment {
 
   class InitializationException(msg: String) extends RuntimeException
 
-  def create(conf: Configuration)(implicit log: Log[Task]): Effect[NodeIdentifier] =
+  def create(conf: Configuration)(implicit log: Log[Task]): Task[NodeIdentifier] =
     for {
-      dataDir <- Task.delay(conf.server.dataDir.toFile).toEffect
+      dataDir <- Task.delay(conf.server.dataDir.toFile)
```
Comment (ArturGajowy) [inline node/src/main/scala/coop/rchain/node/NodeEnvironment.scala]: I'd remove the `.toEffect` calls, but maybe leave the `Effect` type alias and its usage? Just in case we'd want to migrate to ZIO or sth...

OTOH ZIO has a Task type of the same shape, so just the imports would change.
If we were to migrate to a bifunctor, there's no big difference between `Task[A] -> ZIO[E, A]` and `Effect[A] -> Effect[E, A]`

WDYT? When's a type alias worth it?

File: node/src/main/scala/coop/rchain/node/diagnostics/client/Runtime.scala (hunk 2 above)
Comment (rabbitonweb) [reply node/src/main/scala/coop/rchain/node/diagnostics/client/Runtime.scala]: yes, program is task so you no longer call attempt on it. you will not get those `CommErrors` return here.

File: comm/src/main/scala/coop/rchain/comm/rp/HandleMessages.scala (hunk 1 above)
Comment (rabbitonweb) [reply comm/src/main/scala/coop/rchain/comm/rp/HandleMessages.scala]: that gived you back an Either[, X] as a result  which is not what we want, right?

**Comment by rabbitonweb** [comment]
@lukasz-golebiewski I thought about it. The issue here is that people were not using `Effect` instantiation properly. If you look at that last what I had to remove, a lot of those changes were things like `leftT` or `rightT`. I also hope that we will always use just `Task` (or equivalent like like `ZIO`). But I'm happy to to revert last commit.

File: node/src/main/scala/coop/rchain/node/NodeEnvironment.scala (hunk 3 above)
Comment (rabbitonweb) [reply node/src/main/scala/coop/rchain/node/NodeEnvironment.scala]: see my reply to @lukasz-golebiewski

File: comm/src/main/scala/coop/rchain/comm/transport/GrpcTransportServer.scala [hunk 4]
```diff
@@ -137,27 +137,6 @@ class TransportServer(server: GrpcTransportServer) {
...
-        case Right(m) => Task.now(m)
-      }
-    val hb: Blob => Task[Unit] = b =>
-      handleStreamed(b).value.flatMap {
-        case Left(err) =>
-          Log[Task].error(s"Error while handling streamed Packet message. Error: ${err.message}")
-        case Right(_) => Task.unit
-      }
-    EitherT.liftF(start(dis, hb))
-  }
```
Comment (sebtomba) [inline comm/src/main/scala/coop/rchain/comm/transport/GrpcTransportServer.scala]: Please add `attemptAndLog` to `GrpcTransportServer.receive.dispatchInternal` in case `handleStreamed()` fails.
```scala
case msg: StreamMessage =>
  StreamHandler.restore(msg) >>= {
    case Left(ex) =>
      Log[Task].error("Could not restore data from file while handling stream", ex)
    case Right(blob) => handleStreamed(blob)
  }
```

File: node/src/main/scala/coop/rchain/node/NodeEnvironment.scala (hunk 3 above)
Comment (rabbitonweb) [reply node/src/main/scala/coop/rchain/node/NodeEnvironment.scala]: If we do things right chainging the `IO` type should be matter of modifing module `node` - but prooject is not that state yet (see `comm` queue machanism, see casper tests).

Global rename `s/monix.Task/zio.Task` will be least of our concerns

**Comment by rabbitonweb** [comment]
bors try

File: comm/src/main/scala/coop/rchain/comm/rp/HandleMessages.scala (hunk 1 above)
Comment (ArturGajowy) [reply comm/src/main/scala/coop/rchain/comm/rp/HandleMessages.scala]: right, that won't do. Feels awkwardly over-verbose, but having thought more of it, I don't see a way to make it simpler ;/

File: node/src/main/scala/coop/rchain/node/NodeEnvironment.scala (hunk 3 above)
Comment (ArturGajowy) [reply node/src/main/scala/coop/rchain/node/NodeEnvironment.scala]: FWIW it's been quite easy to port one of the tests executing end-to-end through rholang :)
https://github.com/rchain/rchain/compare/dev...ArturGajowy:try_zio

File: comm/src/main/scala/coop/rchain/comm/transport/GrpcTransportServer.scala (hunk 4 above)
Comment (rabbitonweb) [reply comm/src/main/scala/coop/rchain/comm/transport/GrpcTransportServer.scala]: as discussed in call, this was only handling the `CommError` (not throwables) so nothing was removed really.
